### PR TITLE
WIP : feat(timepicker):   Base elements

### LINKED ISSUE #24: Timepicker
Implement timepicker from UI Bootstrap

Should note, this likely will need to be coordinated with the datepicker with the implementation of a default parsing service that can be optionally included that uses the Date constructor.

This should be flexible enough to accept other parsing mechanisms, i.e. moment.js, as long as a common interface is implemented.

### PR DESCRIPTION
**Pull request by sendilkumarn**
Initial Commit 

let me know your comments Thanks 👍 

fixes #24

### CODE REVIEW

File: src/timepicker/timepicker.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+export class NgbTime {
+  constructor(public hour: number, public minute: number, public second: number) {}
+}
+
+@Component({
+  selector: 'ngb-timepicker',
+  exportAs: 'ngbTimepicker',
+  providers: [NGB_TIMEPICKER_VALUE_ACCESSOR],
+  template: `
+    <input [ngModel]="model?.hour" (ngModelChange)="updateHour($event)" [maxLength]=2> <span> : </span>
```
Comment (icfantv) [inline src/timepicker/timepicker.ts]: Should these have BS form field styling?

File: src/timepicker/timepicker.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      this.model.hour = 0;
+    }
+    this.onTouched();
+    this.onChange(this.model);
+  }
+
+  decrementHour() {
+    this.model.hour--;
+    if (this.model.hour < 0) {
+      this.model.hour = 23;
```
Comment (icfantv) [inline src/timepicker/timepicker.ts]: What did we decide on supporting AM/PM?

File: src/timepicker/timepicker.ts (hunk 1 above)
Comment (sendilkumarn) [inline src/timepicker/timepicker.ts]: I will add them

Comment (icfantv) [inline src/timepicker/timepicker.ts]: Input type = number?

File: src/timepicker/timepicker.ts (hunk 2 above)
Comment (sendilkumarn) [inline src/timepicker/timepicker.ts]: This is base skeleton. I will add them

File: src/timepicker/timepicker.ts (hunk 1 above)
Comment (sendilkumarn) [inline src/timepicker/timepicker.ts]: yet to decide on that I guess?

File: src/timepicker/timepicker.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    this.model.second--;
+    if (this.model.second < 0) {
+      this.model.second = 59;
+      this.decrementMinute();
+    }
+    this.propagateModelChange();
+  }
+
+  updateHour(newVal) {
+    let hour = toInteger(newVal);
```
Comment (icfantv) [inline src/timepicker/timepicker.ts]: If we do input type of number is this utility even necessary?  I think there's also the trick of simply pretending the value with a plus sign...

Comment (sendilkumarn) [inline src/timepicker/timepicker.ts]: >  input type of number

that has its own issues. IMO its better to have a normal input type and convert as we need.

> trick of simply pretending the value with a plus sign

we do have a check here, so we can validate with that.

**Comment by pkozlowski-opensource** [comment]
@sendilkumarn thnx for starting this off. I'm going to look at the code, add more tests and see how I can progress. Will probably open a PR with code from this PR and my changes.

**Comment by sendilkumarn** [comment]
Okay then @pkozlowski-opensource let me know if there is anything

**Comment by pkozlowski-opensource** [comment]
Picking up commits from this PR as part of #450 - let's continue the discussion in there.

**Pull request closed without merging**